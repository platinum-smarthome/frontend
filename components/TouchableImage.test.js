import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import TouchAble from './TouchAbleImage'
import { TouchableHighlight, Image } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<TouchAble />)
})

describe('<TouchAble/>', () => {
  it('should render <TouchAble/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<TouchAble/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<TouchableHighlight/>)).toBeDefined()
    expect(wrapper.find('TouchableHighlight').length).toBe(1)  
  })
  it('Should render <Image/>', () => {
    expect(wrapper.find(<Image/>)).toBeDefined()
    expect(wrapper.find('Image').length).toBe(1)  
  })
})

describe('<Image/> renders the correct image', () => {
  it('should render the proper image if status is true', () => {
    expect(wrapper.find(<Image source={require('./assets/info-2-16.png')}/>)).toBeDefined()
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<TouchAble/>)
    expect(tree).toMatchSnapshot()
  })
})