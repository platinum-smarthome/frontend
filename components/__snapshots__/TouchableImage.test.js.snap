// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 50,
      "marginHorizontal": 16,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "testUri": "../../../components/assets/info-2-16.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "height": null,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;
